staging/src/k8s.io/client-go/transport/websocket/roundtripper.go

```diff
@@ -111,9 +113,14 @@ func (rt *RoundTripper) RoundTrip(request *http.Request) (retResp *http.Response
 	wsConn, resp, err := dialer.DialContext(request.Context(), request.URL.String(), request.Header)
 	if err != nil {
 		if errors.Is(err, gwebsocket.ErrBadHandshake) {
```

Comment:
Mimic what SpdyRoundTripper#NewConnection does more closely, trying to read the response into a status error, and not dumping it directly into the output if it is a status error. Something like:

```go
if resp != nil && len(resp.Status) > 0 {
		defer resp.Body.Close()
    var cause error
		responseError := ""
		responseErrorBytes, readErr := io.ReadAll(io.LimitReader(resp.Body, 64*1024))
		if readErr != nil {
			cause = fmt.Errorf("%w (unable to read error from server response)", err)
		} else {
			responseError = strings.TrimSpace(string(responseErrorBytes))
			if obj, _, err := statusCodecs.UniversalDecoder().Decode(responseErrorBytes, nil, &metav1.Status{}); err == nil {
				if status, ok := obj.(*metav1.Status); ok {
					cause = &apierrors.StatusError{ErrStatus: *status}
				}
			}
			if cause == nil {
				cause = fmt.Errorf("%w: %s", err, responseError)
			}
		}
		return nil, &httpstream.UpgradeFailureError{Cause: cause}
```